heron/spi/src/java/com/twitter/heron/spi/utils/NetworkUtils.java

```diff
@@ -142,27 +147,32 @@ public static boolean sendHttpPostRequest(HttpURLConnection connection,
       return false;
     }
 
-    connection.setRequestProperty(CONTENT_TYPE, "application/x-www-form-urlencoded");
-
-    connection.setRequestProperty(CONTENT_LENGTH, Integer.toString(data.length));
+    if (data.length > 0) {
+      connection.setRequestProperty(CONTENT_TYPE, contentType);
+      connection.setRequestProperty(CONTENT_LENGTH, Integer.toString(data.length));
 
-    connection.setUseCaches(false);
-    connection.setDoOutput(true);
+      connection.setUseCaches(false);
+      connection.setDoOutput(true);
 
-    try {
-      connection.getOutputStream().write(data);
-    } catch (IOException e) {
-      LOG.log(Level.SEVERE, "Failed to send request: ", e);
-      return false;
-    } finally {
+      OutputStream os = null;
       try {
-        connection.getOutputStream().close();
+        os = connection.getOutputStream();
+        os.write(data);
+        os.flush();
       } catch (IOException e) {
-        LOG.log(Level.SEVERE, "Failed to close OutputStream: ", e);
+        LOG.log(Level.SEVERE, "Failed to send request: ", e);
         return false;
+      } finally {
+        try {
+          if (os != null) {
+            os.close();
+          }
+        } catch (IOException e) {
+          LOG.log(Level.SEVERE, "Failed to close OutputStream: ", e);
```

Comment:
What is the expectation here for logging this message? It becomes very difficult to know if there was a real issue with a stream which cannot be closed or failed on housekeeping tasks to cleanup resources. Preferably we should just use something like `IOUtils.closeQuietly(os)`